### PR TITLE
refactor: rename `Monoid.{combineAll -> fold}`

### LINKED ISSUE #3027: Rename `Monoid.{combineAll -> fold}`
Rename `Monoid.{combineAll -> fold}` and change its signature and impl to:

```
def fold[t, a](f: f[a]): a with Foldable[f], Monoid[a] = Foldable.fold(f)
```

### PR DESCRIPTION
**Pull request by ninaandrup**
closes #3027

### CODE REVIEW

File: main/src/library/Monoid.flix [hunk 1]
```diff
@@ -37,12 +37,9 @@ pub lawless class Monoid[a] with SemiGroup[a] {
 namespace Monoid {
```
Comment (magnus-madsen) [inline main/src/library/Monoid.flix:37]: Entire namespace can be deleted :)

File: main/test/ca/uwaterloo/flix/library/TestMonoid.flix [hunk 2]
```diff
@@ -17,29 +17,29 @@
 namespace TestMonoid {
 
     use Monoid.{empty, combine};
-    use Monoid.{combineAll};
+    use Monoid.{fold};
 
     /////////////////////////////////////////////////////////////////////////////
-    // combineAll                                                              //
+    // fold                                                                    //
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestMonoid.flix:23]: Do we not already have tests for fold?

Comment (ninaandrup) [reply main/test/ca/uwaterloo/flix/library/TestMonoid.flix:23]: Yes, in `TestFoldable.flix`. Should I just delete them here?

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestMonoid.flix:23]: Yes, delete them.

**Pull request merged**